tools/mtouch/Target.cs

```diff
@@ -1439,17 +1439,17 @@ public void NativeLink (BuildTasks build_tasks)
 			build_tasks.Add (link_task);
 		}
 
-		public void AdjustDylibs ()
+		public static void AdjustDylibs (string output)
 		{
 			var sb = new StringBuilder ();
-			foreach (var dependency in Xamarin.MachO.GetNativeDependencies (Executable)) {
+			foreach (var dependency in Xamarin.MachO.GetNativeDependencies (output)) {
```

Comment:
What happens if `output` is `null` here? Does `Xamarin.MachO.GetNativeDependencies` always return at least 1 element?

Reply:
> What happens if output is null here? 

`output` can't be `null` - it's either the `Executable` or (for Xamarin.Sdk shared framework) the `OutputFile`

if they are not built (which does not mean `output` would be `null`) then an error would be reported before reaching this point

if is was called by a new/future callsite then an `ArgumentNullException` would be thrown by `GetNativeDependencies` when looking up the dictionary with `TryGetValue`, in turn this would result in an `MT0000` error

> Does Xamarin.MachO.GetNativeDependencies always return at least 1 element?

it should, but even if it did not it still would not return `null` but an empty `HashSet<string>` and the execution would be fine